crates/forge_loader/src/manifest.rs

```diff
@@ -208,63 +401,239 @@ impl<T> AsRef<T> for FunctionTy<T> {
 }
 
 impl<'a> ForgeModules<'a> {
-    pub fn into_analyzable_functions(
-        mut self,
-    ) -> impl Iterator<Item = FunctionTy<FunctionMod<'a>>> {
-        //FIXME: The logic here is incorrect, a function should be in the ignored function IIF it
-        //is not invoked by a user-invocable module.
+    // TODO: function returns iterator where each item is some specified type.
+    pub fn into_analyzable_functions(self) -> impl Iterator<Item = Entrypoint<'a>> {
+        // destructuring ForgeModules to remember to add new modules to this
+        let Self {
+            mut webtriggers,
+            custom_field,
+            consumers: _,
+            functions,
+            event_triggers: _,
+            scheduled_triggers: _,
+            compass_admin_page,
+            component_page,
+            data_provider,
```

Comment:
After looking at the [data provider](https://developer.atlassian.com/platform/forge/manifest-reference/modules/compass-data-provider/) docs, I think this might not be invokable, since it can only receive [events](https://developer.atlassian.com/cloud/compass/integrations/create-a-data-provider-app/). There's no user context in the examples either, so we can probably treat this module as non invokable.